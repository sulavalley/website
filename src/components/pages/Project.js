import React from 'react';
import { string } from 'prop-types';
import { Card, Image } from 'semantic-ui-react';

class Project extends React.Component {
  render() {
    const { name , url, description } = this.props;
 
    return (
      <Card>
        <Image
          src={url}
          size="medium"
        />
        <Card.Content>
          <Card.Header>{name}</Card.Header>
          <Card.Description>
            {description}
          </Card.Description>
        </Card.Content>
      </Card>
    );
  }
}

Project.propTypes = {
  name: string,
  url: string,
  description: string
};

export default Project;
