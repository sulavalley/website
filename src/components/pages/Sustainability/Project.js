import React from 'react';
import { Item } from 'semantic-ui-react';

class Project extends React.Component {
  render() {
    const { name, url, description } = this.props;

    return (
      <Item>
        <Item.Image src={url} size="small" />
        <Item.Content>
          <Item.Header>{name}</Item.Header>
          <Item.Description>{description}</Item.Description>
        </Item.Content>
      </Item>
    );
  }
}

export default Project;
