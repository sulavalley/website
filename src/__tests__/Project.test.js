import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Project from './../components/pages/Sustainability/Project';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Project />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  shallow(<Project />);
});

test('renders correctly', () => {
  const tree = renderer.create(<Project />).toJSON();
  expect(tree).toMatchSnapshot();
});
