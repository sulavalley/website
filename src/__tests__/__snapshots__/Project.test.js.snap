// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}
>
  <img
    className="ui medium image"
    src={undefined}
  />
  <div
    className="content"
  >
    <div
      className="header"
    />
    <div
      className="description"
    />
  </div>
</div>
`;
